test/functional/legacy/094_visual_mode_operators_spec.lua

```diff
@@ -0,0 +1,165 @@
+      xxxxxxxxxxxxx
+      xxxxxxxxxxxxx]])
+
+    -- Exercise characterwise Visual mode plus operator, with count and repeat.
+    execute('/^apple')
+    feed('lvld.l3vd.')
+
+    -- Same in linewise Visual mode.
+    execute('/^line 1')
+    feed('Vcnewline<esc>j.j2Vd.')
+
+    -- Same in blockwise Visual mode.
+    execute('/^xxxx')
+    feed('jlc  <esc>l.l2c----<esc>l.')
+
+    -- Assert buffer contents.
+    expect([[
+      a y
+      
+      newline
+      newline
+      
+          --------x
+          --------x
+      xxxx--------x
+      xxxx--------x]])
+  end)
+end)
+
+describe('Visual mode mapping', function()
+  setup(function()
+    clear()
+    source_user_functions()
+  end)
+
+  it('is working', function()
+    insert([[
+      KiwiRaspberryDateWatermelonPeach
+      JambuRambutanBananaTangerineMango]])
+
+    -- Set up Visual mode mappings.
+    execute('vnoremap W /\\u/s-1<CR>')
+    execute('vnoremap iW :<C-U>call SelectInCaps()<CR>')
+
+    -- Do a simple change using the simple vmap, also with count and repeat.
+    execute('/^Kiwi')
+    feed('vWcNo<esc>l.fD2vd.')
+
+    -- Same, using the vmap that maps to an Ex command.
+    execute('/^Jambu')
+    feed('llviWc-<esc>l.l2vdl.')
+
+    -- Assert buffer contents.
+    expect([[
+      NoNoberryach
+      --ago]])
+  end)
+end)
+
+describe('Operator-pending mode mapping', function()
+  setup(function()
+    clear()
+    source_user_functions()
+  end)
+
+  it('is working', function()
+    insert([[
+      PineappleQuinceLoganberryOrangeGrapefruitKiwiZ
+      JuniperDurianZ
+      LemonNectarineZ]])
+
+    -- Set up Operator-pending mode mappings.
+    execute('onoremap W /\\u/<CR>')
+    execute('onoremap <Leader>W :<C-U>call MoveToCap()<CR>')
+    execute('onoremap iW :<C-U>call SelectInCaps()<CR>')
+
+    -- Do a simple change using the simple omap, also with count and repeat.
+    execute('/^Pineapple')
+    feed('cW-<esc>l.l2.l.')
+
+    -- Same, using the omap that maps to an Ex command to move the cursor.
+    execute('/^Juniper')
+    feed('g?\\WfD.')
+
+    -- Same, using the omap that uses Ex and Visual mode (custom text object).
+    execute('/^Lemon')
+    feed('yiWPlciWNew<esc>fr.')
+
+    -- Assert buffer contents.
+    expect([[
+      ----Z
+      WhavcreQhevnaZ
+      LemonNewNewZ]])
+  end)
+end)
+
+describe('patch 7.3.879', function()
+  setup(clear)
+
+  it('is working', function()
+    insert([[
+      zzzz
+      zzzz]])
+
+    -- Start a change operation consisting of operator plus Ex command, like
+    -- "dV:..." etc., then either
+    -- - complete the operation by pressing Enter: as a result the buffer is
+    --   changed, taking into account the v/V/CTRL-V modifier given; or
+    -- - abort the operation by pressing Escape: no change to the buffer is
+    --   carried out.
+    execute('/^zzzz')
+    feed([[dV:
dv:
:set noma | let v:errmsg = ''<cr>]])
+    feed([[d:
:set ma | put = v:errmsg =~# '^E21' ? 'ok' : 'failed'<cr>]])
```

Comment:
You should be able to indent the `feed` blocks, same as `insert` or `expect`. Common prefixed whitespace will be [removed](https://github.com/neovim/neovim/blob/master/test/functional/helpers.lua#L155).